cddl/contrib/opensolaris/lib/libdtrace/common/dt_module.c

```diff
@@ -815,6 +803,19 @@ dt_module_load(dtrace_hdl_t *dtp, dt_module_t *dmp)
 	dmp->dm_strtab.cts_entsize = 0;
 	dmp->dm_strtab.cts_offset = 0;
 
+	// TODO: remove this when we are able to load the kernel symtable
+#ifdef __mips__
+	if (strcmp(dmp->dm_name,"kernel") == 0) {
+		dt_dprintf(
+		    "LOADING .dynsym INSTEAD OF .symtab FOR MIPS! "
+		    "A lot of symbols will be missing\n");
+		dmp->dm_symtab.cts_name = ".dynsym";
+		dmp->dm_symtab.cts_type = SHT_DYNSYM;
+		// The string table for .dynsym is .dynstr, not .strtab.
+		dmp->dm_strtab.cts_name = ".dynstr";
```

Comment:
Do you not also need `dmp->dm_strtab.cts_type = SHT_DYNSYM;`?

Reply:
nope, the type of `dynstr` is still STRTAB
```
  [ 6] .dynstr           STRTAB           ffffffff802dcd88  000dcd88                                                                               
       000000000006e45f  0000000000000000   A       0     0     1   
```

Reply:
Oh, yes, I misread this as `.dynsym`.